br/pkg/lightning/restore/check_info.go

```diff
@@ -66,252 +46,69 @@ func (rc *Controller) isSourceInLocal() bool {
 	return strings.HasPrefix(rc.store.URI(), storage.LocalURIPrefix)
 }
 
-func (rc *Controller) getReplicaCount(ctx context.Context) (uint64, error) {
-	replConfig, err := rc.preInfoGetter.GetReplicationConfig(ctx)
-	if err != nil {
-		return 0, errors.Trace(err)
+// clusterResource check cluster has enough resource to import data. this test can by skipped.
+func (rc *Controller) clusterResource(ctx context.Context) error {
+	checkCtx := ctx
+	if rc.taskMgr != nil {
+		checkCtx = WithPrecheckKey(ctx, taskManagerKey, rc.taskMgr)
 	}
-	return replConfig.MaxReplicas, nil
-}
-
-func (rc *Controller) getClusterAvail(ctx context.Context) (uint64, error) {
-	storeInfo, err := rc.preInfoGetter.GetStorageInfo(ctx)
+	theChecker, err := rc.precheckItemBuilder.BuildPrecheckItem(CheckTargetClusterSize)
```

Comment:
code since this line is quite similar with other methods, we can abstract into a helper method.